file-source/src/main/java/org/springframework/cloud/stream/module/file/source/FileSource.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.integration.dsl.file.Files;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.file.FileReadingMessageSource;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.file.transformer.FileToByteArrayTransformer;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.scheduling.Trigger;
+import org.springframework.util.StringUtils;
+
+/**
+ * Creates a {@link FileReadingMessageSource} bean and registers it as a
+ * Inbound Channel Adapter that sends messages from the Source output channel.
+ *
+ * @author Gary Russell
+ */
+@EnableBinding(Source.class)
+@Import({PeriodicTriggerConfiguration.class, SpelExpressionConverterConfiguration.class})
+@EnableConfigurationProperties({ FileSourceProperties.class,
+	FileConsumerProperties.class, MaxMessagesProperties.class})
+public class FileSource {
+
+	@Autowired
+	private FileSourceProperties properties;
+
+	@Autowired
+	private FileConsumerProperties fileConsumerProperties;
+
+	@Autowired
+	MaxMessagesProperties maxMessagesProperties;
+
+	@Autowired
+	Trigger trigger;
+
+	@Bean
+	public PollerMetadata poller() {
+		return Pollers.trigger(trigger).maxMessagesPerPoll(this.maxMessagesProperties.getMaxMessages()).get();
+	}
+
+	@Autowired
+	Source source;
+
+	@Bean
+	public IntegrationFlow fileSourceFlow() {
+		FileInboundChannelAdapterSpec messageSourceSpec = Files.inboundAdapter(new File(this.properties.getDirectory()));
+
+		if (StringUtils.hasText(this.properties.getFilenamePattern())) {
+			messageSourceSpec.patternFilter(this.properties.getFilenamePattern(), this.properties.isPreventDuplicates());
+		}
+		else if (this.properties.getFilenameRegex() != null) {
+			messageSourceSpec.regexFilter(this.properties.getFilenameRegex(), this.properties.isPreventDuplicates());
+		}
+
+		IntegrationFlowBuilder flowBuilder = IntegrationFlows
+				.from(messageSourceSpec,
+					new Consumer<SourcePollingChannelAdapterSpec>() {
+
+						@Override
+						public void accept(SourcePollingChannelAdapterSpec sourcePollingChannelAdapterSpec) {
+							sourcePollingChannelAdapterSpec
+									.autoStartup(false)
+									.poller(poller());
+						}
+
+					});
+		return enhanceFlowForReadingMode(flowBuilder)
+			.channel(source.output())
+			.get();
+	}
+
+	private IntegrationFlowBuilder enhanceFlowForReadingMode(IntegrationFlowBuilder flowBuilder) {
+		switch (this.fileConsumerProperties.getMode()) {
+		case contents:
+			flowBuilder.enrichHeaders(Collections.<String, Object>singletonMap(MessageHeaders.CONTENT_TYPE,
+					"application/octet-stream"))
+					.transform(new FileToByteArrayTransformer());
+			break;
+		case lines:
+			flowBuilder.enrichHeaders(Collections.<String, Object>singletonMap(MessageHeaders.CONTENT_TYPE,
+					"text/plain"))
```

Comment:
What I'm saying is: we don't know that it is `plain`. Just because we split on newlines, doesn't mean that it's not, say, `text/xml`. Not a big deal, as we can later force the output type. But may be too big of an assumption